### PR TITLE
ICU-21051 Add Locale::createCanonical performance test

### PR DESCRIPTION
**Pull request by FrankYFTang**
##### Checklist

- [X] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21051
- [X] Updated PR title and link in previous line to include Issue number
- [X] Issue accepted
- [X] Tests included
- [X] Documentation is changed or added

This is the first step toward fixing the performance issue- by adding a simple performance test so we can measure the two builds.

### CODE REVIEW

File: icu4c/source/test/perf/localecanperf/localecanperf.cpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+// Test case ...
+//
+class LocaleCreateCanonical : public UPerfFunction {
+public:
+    LocaleCreateCanonical() {
+        testCases.push_back("en");
+        testCases.push_back("en-US");
+        testCases.push_back("ja-JP");
+        testCases.push_back("zh-Hant-CN");
+        testCases.push_back("hy-SU");
```
Comment (sffc) [inline icu4c/source/test/perf/localecanperf/localecanperf.cpp:25]: Do you want to include more interesting test cases, like ones that include unicode keywords, or that have different canonical forms, like maybe iw/he?

Comment (FrankYFTang) [reply icu4c/source/test/perf/localecanperf/localecanperf.cpp:25]: will do that later.

**Comment by hugovdm** [comment]
Copyright check is complaining:

```
$ perl tools/scripts/cpysearch/cpyscan.pl
icu4c/source/test/perf/localecanperf/runTest.sh
Above files did not contain the correct copyright notice. at tools/scripts/cpysearch/cpyscan.pl line 54.
The command "perl tools/scripts/cpysearch/cpyscan.pl" exited with 255.
```

**Comment by markusicu** [comment]
> Travis failure: Copyright check is complaining:
> 
> ```
> $ perl tools/scripts/cpysearch/cpyscan.pl
> icu4c/source/test/perf/localecanperf/runTest.sh
> Above files did not contain the correct copyright notice. at tools/scripts/cpysearch/cpyscan.pl line 54.
> The command "perl tools/scripts/cpysearch/cpyscan.pl" exited with 255.
> ```

Yes, and it's showing up as a test failure on pending PRs.
@FrankYFTang please send another PR adding the copyright lines.

**Comment by markusicu** [comment]
> @FrankYFTang please send another PR adding the copyright lines.

--> PR #1281

File: icu4c/source/configure [hunk 2]
```diff
@@ -8762,6 +8762,7 @@ do
     "test/perf/ubrkperf/Makefile") CONFIG_FILES="$CONFIG_FILES test/perf/ubrkperf/Makefile" ;;
     "test/perf/charperf/Makefile") CONFIG_FILES="$CONFIG_FILES test/perf/charperf/Makefile" ;;
     "test/perf/convperf/Makefile") CONFIG_FILES="$CONFIG_FILES test/perf/convperf/Makefile" ;;
+    "test/perf/localecanperf/Makefile") CONFIG_FILES="$CONFIG_FILES test/perf/localecanperf/Makefile" ;;
```
Comment (srl295) [inline icu4c/source/configure:8765]: @FrankYFTang I assume that `configure.ac` was missed in the checkin, or this file was updated manually.
